### PR TITLE
Feature/GitHub issue data for service

### LINKED ISSUE #155: Store Github issue data for a service
Create collection, schema and publication for the following Github issues for a service. 

Blocks #144.

Definition of done
---
* [x] Collection for Github issues exists.
* [x] Schema for Github issues exists.
* [x] Schema mirrors [Github API structure](https://developer.github.com/v3/issues)
* [x] When adding a Github repository to a service, all issues are pulled to the collection.
* [x] Publication for Github issues exists.

### PR DESCRIPTION
**Pull request by frenchbread**
Closes #155 

# Proposed changes
- Added `GithubIssues` Collection
- Added `githubGetRepoIssues` and `githubSyncRepoIssues` methods
- Changed `integrations` folder structure. Moved global github object init to separate folder
- Autoform hook which loads all issues for provided repo into local db.

@Digipalvelutehdas/dipor-team please review.

### CODE REVIEW

File: core/server/github.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import GithubApi from "github";
```
Comment (brylie) [inline core/server/github.js]: This was added in the `/integrations/github/...` namespace.

File: integrations/github/issues/server/methods.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+Meteor.methods({
+  githubGetRepoIssues (user, repo, page, perPage) {
```
Comment (brylie) [inline integrations/github/issues/server/methods.js]: Give the `page` and `perPage` arguments default values:

- page=1
- perPage=100

File: integrations/github/issues/server/methods.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    @param {string} user - The Github username or organization.
+    @param {string} repo - The Github repository name associated with the user.
+    */
+
+    const repoIssues = Async.runSync((done) => {
+      Github.issues.repoIssues({
+        user: user,
+        repo: repo,
+        state: 'all',
+        page: page || 1, // Default to the first page
```
Comment (brylie) [inline integrations/github/issues/server/methods.js]: Move the default parameters to the function definition.

File: integrations/github/issues/server/publications.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/* Repository Issues */
+Meteor.publish("repositoryIssues", (repo) => {
```
Comment (brylie) [inline integrations/github/issues/server/publications.js]: Name this parameter `repository`.

File: integrations/github/issues/server/publications.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+
+  // Find all Issues for repo
+  const repositoryIssues = GithubIssues.find({ repository: repo });
```
Comment (brylie) [inline integrations/github/issues/server/publications.js]: When you rename the function parameter to `repository`, you can use ES2015 enhanced object literal for the query parameter:

```
... GithubIssues.find({ repository });

// The above is equivalent to:
// GithubIssues.find({ repository: repository });
```

File: sources/client/add/autoform.js [hunk 6]
```diff
@@ -1,5 +1,19 @@
 AutoForm.addHooks("addSourceForm", {
-  "onSuccess": function () {
+  before: {
+    insert (source) {
+
+      // Check if selected source is 'github'
+      if (source.type === 'github') {
+
+        // Get repository issues list for provided user/organization and repo
+        Meteor.call("githubSyncRepoIssues", source.github.user, source.github.repo);
```
Comment (brylie) [inline sources/client/add/autoform.js]: This sync job should happen in the `onSuccess` callback, so the user does not have to wait.

**Comment by brylie** [comment]
There are a few changes here. In general, please review the work in the `feature/commits-over-time` branch. That way the general ideas in each branch mirror one another.

**Comment by frenchbread** [comment]
Oh man! I've run into some issues after changing code to ES6 look. And I thought that errors pop up because of it. It turns out that there is such thing as [rate-limiting](https://developer.github.com/v3/#rate-limiting) for Github API. This allows to make only 60 requests per hour if not using authentication. We should keep that in mind when doing a `synced-cron` feature. Because if mirroring a repo that happened to have a lot of issues (or commits), we run out of requests pretty quickly.

**Comment by frenchbread** [comment]
Added suggested changes, please review. I've also added error handling for synchronous callback.

**Comment by brylie** [comment]
@frenchbread, @Alapan will review this PR, since you both are working on similar tasks.

**Pull request merged**